cmd/erasure-server-pool.go

```diff
@@ -244,6 +244,41 @@ func (z *erasureServerPools) getServerPoolsAvailableSpace(ctx context.Context, s
 	return serverPools
 }
 
+// getPoolIdxExisting returns the (first) found object pool index containing an object.
+// If the object exists, but the latest version is a delete marker, the index with it is still returned.
+// If the object does not exist ObjectNotFound error is returned.
+// If any other error is found, it is returned.
+// The check is skipped if there is only one zone, and 0, nil is always returned in that case.
+func (z *erasureServerPools) getPoolIdxExisting(ctx context.Context, bucket, object string) (idx int, err error) {
+	if z.SinglePool() {
+		return 0, nil
+	}
+
+	errs := make([]error, len(z.serverPools))
+	objInfos := make([]ObjectInfo, len(z.serverPools))
+
+	var wg sync.WaitGroup
+	for i, pool := range z.serverPools {
+		wg.Add(1)
+		go func(i int, pool *erasureSets) {
+			defer wg.Done()
+			objInfos[i], errs[i] = pool.GetObjectInfo(ctx, bucket, object, ObjectOptions{})
+		}(i, pool)
+	}
+	wg.Wait()
+
+	for i, err := range errs {
+		if err == nil {
+			return i, nil
+		}
+		if !isErrObjectNotFound(err) {
+			return -1, err
+		}
+	}
```

Comment:
You need to keep this otherwise this will bring regression for server side replication.

```go
        for i, err := range errs {
                if err != nil && !isErrObjectNotFound(err) {
                        return -1, err
                }
                if isErrObjectNotFound(err) {
                        // No object exists or its a delete marker,
                        // check objInfo to confirm.
                        if objInfos[i].DeleteMarker && objInfos[i].Name != "" {
                                return i, nil
                        }
                        // objInfo is not valid, truly the object doesn't
                        // exist proceed to next pool.
                        continue
                }
                // object exists at this pool.
                return i, nil
        }
```